transport/websocketpeer.go

```diff
@@ -115,7 +137,12 @@ func ConnectWebsocketPeer(routerURL string, serialization serialize.Serializatio
 		dialer.EnableCompression = true
 	}
 
-	conn, _, err := dialer.Dial(routerURL, nil)
+	if ctx == nil {
```

Comment:
Allowing a nil context instead of using `context.Background()` does not save anything, since `dialer.Dial()` just ends up calling `dialer.DialContext(context.Background(),..)` anyway:
https://golang.org/src/net/dial.go?s=10483:10543#L329